File: src/main/java/org/oddjob/dido/UnsupportedDataInException.java
package org.oddjob.dido;

/**
 * Something has gone wrong with reading or writing data.
 * 
 * @author rob
 *
 */
public class UnsupportedDataInException extends DataException {
	private static final long serialVersionUID = 2010072700L;
	
	
	public UnsupportedDataInException(Class<? extends DataIn> provider, 
			Class<? extends DataIn> required) {
		super(provider.getName() + " cannot provide " + required.getName());
	}

}

File: src/main/scala/org/oddjob/dido/UnsupportedDataInException.scala
package org.oddjob.dido

/**
 * Something has gone wrong with reading or writing data.
 * 
 * @author rob
 *
 */
class UnsupportedDataInException(provider: Class[_ <: DataIn], required: Class[_ <: DataIn])
        extends DataException(provider.getName() + " cannot provide " + required.getName()) {

	private val serialVersionUID = 2010072700L

}
